mindmap/mindmap.py

```diff
@@ -90,20 +135,97 @@ def studio_view(self, context=None):
         frag.initialize_js("MindMapXBlock")
         return frag
 
-    # TO-DO: change this handler to perform your own actions.  You may need more
-    # than one handler, or you may not need any handlers at all.
-    @XBlock.json_handler
-    def increment_count(self, data, suffix=''):
+    @staticmethod
+    def connect_to_s3() -> Tuple[boto3.client, str]:
+        """
+        Create a connection to S3.
+
+        Raises:
+            MisconfiguredMindMapService: If the AWS settings are not configured.
+
+        Returns:
+            Tuple[boto3.client, str]: The S3 client and the bucket name.
+        """
+        aws_access_key_id = getattr(settings, "AWS_ACCESS_KEY_ID", None)
+        aws_secret_access_key = getattr(settings, "AWS_SECRET_ACCESS_KEY", None)
+        aws_bucket_name = getattr(settings, "AWS_BUCKET_NAME", None)
+
+        if not aws_access_key_id or not aws_secret_access_key or not aws_bucket_name:
+            raise MisconfiguredMindMapService("AWS settings not configured")
+
+        s3_client = boto3.client(
+            "s3",
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+        )
+
+        return s3_client, aws_bucket_name
+
+    def get_file_key(self, anonymous_user_id: str) -> str:
+        """
+        Return the key (path) to save and retrieve the file in S3.
+
+        Returns:
+            str: The key (path) to use in S3.
         """
-        An example handler, which increments the data.
+        block_id = self.location.block_id # pylint: disable=no-member
+        return f"{block_id}/{anonymous_user_id}/mindmap.json"
+
+    def file_exists_in_s3(self, anonymous_user_id: str) -> bool:
+        """
+        Check if the file exists in S3.
+
+        Returns:
+            bool: True if the file exists, False otherwise.
         """
-        if suffix:
-            pass  # TO-DO: Use the suffix when storing data.
-        # Just to show data coming in...
-        assert data['hello'] == 'world'
+        s3_client, aws_bucket_name = self.connect_to_s3()
+        try:
+            s3_client.head_object(
+                Bucket=aws_bucket_name,
+                Key=self.get_file_key(anonymous_user_id)
+            )
+        except s3_client.exceptions.ClientError as error:
+            if error.response["Error"]["Code"] == str(HTTPStatus.NOT_FOUND):
+                return False
+            raise S3ClientError(error) from error
```

Comment:
why can't why reraise the same error `raise error`? That way we don't need to create a new exception